src/app.ts

```diff
@@ -292,67 +347,19 @@ function optimizeRow(
-      "impressions_7d": 13,
-      "clicks_7d": 0
-    }
-
-    ${ORIGINAL_TITLE_TEMPLATE_PROMPT} brand ${SEPARATOR} model ${SEPARATOR} product
-    ${CATEGORY_PROMPT} Guitars
-    ${TEMPLATE_PROMPT} brand ${SEPARATOR} model ${SEPARATOR} product ${SEPARATOR} color ${SEPARATOR} design
-    ${ATTRIBUTES_PROMPT} Seymour Duncan ${SEPARATOR} SM-1 ${SEPARATOR} Mini Humbucker ${SEPARATOR} Chrome ${SEPARATOR} Vintage
-    ${TITLE_PROMPT} Seymour Duncan SM-1 Mini Humbucker Pickup in Chrome
-
-    Context:
-    {
-      "item_id": "565119",
-      "title": "Gretsch Drums 14\"\"x7\"\" 140th Anniversary Snare",
-      "description": "Gretsch Drums 140th Anniversary Snare + Bag; Farbe: Natur; Hochglanz lackiert; Ahorn Kessel; 16 Einzelböckchen; Nickel Hardware; Figured Ash Außenlage; Micro Sensitive Anhebung; Snap-in Stimmschlüsselhalter; inkluseive 140th Anniversary Tasche; Zertifikat unterzeichnet von allen Produktionsmitarbeitern; 140 Stück limitiert",
-      "brand": "Gretsch Drums",
-      "color": "Natur",
-      "product_type": "Drums und Percussion > Akustik-Drums > Snaredrums mit Holzkessel > 14\"\" Holz Snaredrums",
-      "impressions_7d": 84,
-      "clicks_7d": 1
-    }
-
-    ${ORIGINAL_TITLE_TEMPLATE_PROMPT} brand ${SEPARATOR} size ${SEPARATOR} edition ${SEPARATOR} product
-    ${CATEGORY_PROMPT} Drums
-    ${TEMPLATE_PROMPT} brand ${SEPARATOR} product ${SEPARATOR} material ${SEPARATOR} edition ${SEPARATOR} size
-    ${ATTRIBUTES_PROMPT} Gretsch Drums ${SEPARATOR} Snare + Bag ${SEPARATOR} Ahorn ${SEPARATOR} 140th Anniversary ${SEPARATOR} 14"x7"
-    ${TITLE_PROMPT} Gretsch Snare Drums + Bag aus Ahorn, 140th Anniversary edition
-
-    Context:
-    {
-      "item_id": "302293",
-      "title": "Thon CD Player Case American Audio",
-      "description": "Thon CD Player Case American Audio, maßgefertigtes Haubencase für American Audio Radius 1000, 2000 und 3000, aus 7mm Multiplex, 25 x 25 mm Alukante, Schaumstoffpolsterung, 2x Butterfly-Verschlüsse, 1 Koffergriff, Gummifüße, Platz für Netzkabel, Stahlkugelecken, hergestellt in Deutschland, Außenmaße (BxTxH): ca. 34 x 50 x 19,4 cm, Gewicht: ca. 4,9 kg, Gewicht mit Radius: ca. 8,7 kg",
-      "brand": "Thon",
-      "color": "Phenol Braun",
-      "product_type": "DJ-Equipment > Zubehör für DJs > DJ Player Cases/Bags",
-      "impressions_7d": 15,
-      "clicks_7d": 0
-    }
-
-    ${ORIGINAL_TITLE_TEMPLATE_PROMPT} brand ${SEPARATOR} product ${SEPARATOR} compatibility
-    ${CATEGORY_PROMPT} DJ Equimpent
-    ${TEMPLATE_PROMPT} brand ${SEPARATOR} product ${SEPARATOR} weight ${SEPARATOR} compatibility ${SEPARATOR} color
-    ${ATTRIBUTES_PROMPT} Thon ${SEPARATOR} CD Player Case ${SEPARATOR} 4,9 kg ${SEPARATOR} American Audio ${SEPARATOR} braun
-    ${TITLE_PROMPT} Thon CD PlayerCase, 4,9 kg, American Audio, braun
-
-    Context:
-    ${JSON.stringify(data, null, 2)}
-
-    `;
-
+function generateEntry(data: Record<string, unknown>): string {
```

Comment:
I'd keep this as "generateTitle". Indeed we generate way more than merely a title but it would be more understandable this way when we add a "generateDescription" later on, which will be with another prompt anyway and therefore cannot be done in this same "generateEntry" function.

Reply:
fetchTitleGenerationData ?